src/pudl/extract/ferc714.py

```diff
@@ -88,19 +136,80 @@ def _extract_raw_ferc714(context):
         )
         with (
             ds.get_zipfile_resource("ferc714", name="ferc714.zip") as zf,
-            zf.open(FERC714_FILES[table_name]["name"]) as csv_file,
+            zf.open(FERC714_CSV_FILES[table_name]["name"]) as csv_file,
         ):
             df = pd.read_csv(
                 csv_file,
-                encoding=FERC714_FILES[table_name]["encoding"],
+                encoding=FERC714_CSV_FILES[table_name]["encoding"],
             )
         if table_name != "respondent_id":
             df = df.query("report_yr in @ferc714_settings.years")
         return df
 
-    return _extract_raw_ferc714
+    return _extract_raw_ferc714_csv
+
+
+@asset
+def raw_ferc714_xbrl__metadata_json(
+    context,
+) -> dict[str, dict[str, list[dict[str, Any]]]]:
+    """Extract the FERC 714 XBRL Taxonomy metadata we've stored as JSON.
+
+    Returns:
+        A dictionary keyed by PUDL table name, with an instant and a duration entry
+        for each table, corresponding to the metadata for each of the respective instant
+        or duration tables from XBRL if they exist. Table metadata is returned as a list
+        of dictionaries, each of which can be interpreted as a row in a tabular
+        structure, with each row annotating a separate XBRL concept from the FERC 714
+        filings.
+    """
+    metadata_path = PudlPaths().output_dir / "ferc714_xbrl_taxonomy_metadata.json"
+    with Path.open(metadata_path) as f:
+        xbrl_meta_all = json.load(f)
+
+    def get_metadata(xbrl_table: str | list[str], xbrl_meta_all):
+        if isinstance(xbrl_table, str):
+            xbrl_table = [xbrl_table]
+        return [
+            metadata
+            for table_name in xbrl_table
+            for metadata in xbrl_meta_all.get(table_name, [])
+            if metadata
+        ]
+
+    xbrl_meta_out = copy.deepcopy(FERC714_XBRL_FILES)
+    for table_name, periods in FERC714_XBRL_FILES.items():
+        for period, tt in periods.items():
+            xbrl_meta_out[table_name][period] = get_metadata(tt, xbrl_meta_all)
 
+    return xbrl_meta_out
 
-raw_ferc714_assets = [
-    raw_ferc714_asset_factory(table_name) for table_name in FERC714_FILES
+
+def create_raw_ferc714_xbrl_assets() -> list[SourceAsset]:
+    """Create SourceAssets for raw FERC 714 XBRL tables.
+
+    SourceAssets allow you to access assets that are generated elsewhere.
+    In our case, the XBRL database contains the raw FERC 714 assets from
+    2021 onward. Prior to that, the assets are distributed as CSVs and
+    are extracted with the ``raw_ferc714_csv_asset_factory`` function.
+
+    Returns:
+        A list of FERC 714 SourceAssets.
+    """
```

Comment:
I'm not sure how to test whether this works because it doesn't create assets, rather references to non-Dagster tables.

Reply:
```python
from pudl.extract.ferc714 import FERC714_XBRL_FILES
raw_table_names = [
        v for value in FERC714_XBRL_FILES.values() for v in value.values()
    ]
defs.load_asset_value(f"raw_ferc714_xbrl__{raw_table_names[0]}")
```